app/controllers/users_controller.rb

```diff
@@ -1,3 +1,28 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
+    if @user.blank?
+      redirect_to signup_path
+    end
```

Comment:
đổi line code từ 8 -> 10 thành như bên dưới nhé 
`return if @user`
`flash[:waring] = " user not found"`
`redirect_to root_path`